### ISSUE TITLE
Dev: Send emails when someone is added as a domain manager

### ISSUE THREAD

**Issue by Katherine-Osos** [cisagov/manage.get.gov#2370]
### Story

As a domain manager 
I want to know when managers have been **added** to my domains
so that I'm aware who has access to those domains.

### Acceptance Criteria

- [x] This new [Email is sent](https://docs.google.com/document/d/1P8WUqPdggGHjocTCktXgZDcJI-CiUY5sEj5Na2loph0/edit) to all managers of a domain when someone is **invited** to manage the domain. 
  - [x]   **Registrar:** This new email should be sent if the person was invited via the registrar
    - Note: we send invites in the registrar from 1) Member management- domain assignment flow and 2) the Domain pages under Domain Managers. Both should be included.
    - Note: This applies when someone is added to a domain via the registrar but they are not yet a user in our system (aka a _domain invitatio_n is made) AND it also applies when someone is added to a domain via the registrar while being an existing user- aka a _UserDomainRole_ is created
  - [x] **Django admin:** This new email should be sent if the person was invited via the django admin DomainInvitation table.
    - [x]  It should **not** be sent if user was added via the User Domain Roles table (yes this is intentionally different then the registrar set up, ask a lead if you have questions)
    - [x] In addition to this new email, we also need to send an email to the person who was invited. That's an [existing email ](https://docs.google.com/document/d/18eQRA5QWu5Ct6ICK2-FCZJf9kbbk4LklekRRfb6g9tQ/edit?tab=t.0#heading=h.gjdgxs) that should have been added to domain invitations- django admin in #2960. Ensure the ACs in #2960 are maintained
- [x] this email applies to org model and non org model equally
#### Add static messaging in Django:
- [x] Add a static message to the top of the "**Add Domain invitation**" page  that reads: "If you add someone to a domain here, it will trigger emails to the invitee and all managers of the domain when you click "save." If you don't want to trigger those emails, use the User domain roles permissions table instead."
  - [x] Link to the "User Domain Roles" table in that message.
- [x] Add a static message to the top of the "**Add user domain role**" page that reads: "If you add someone to a domain here, it won't trigger any emails. To trigger emails, use the User Domain Role invitations table instead."
  - [x] Link to the "Portfolio invitations" table in that message.
- [x]  Use slim blue alert, placed right below the heading on the "add" page for those tables.

### Additional Context

- This issue very similar to #3285, whoever does this issue should consider picking up 3285 as well.
* This issue is about providing user notifications. It might seem like m5 is the wrong place, since m7 is about notifications – but from a product perspective, we want this in m5 **so that members can actually be managed** and get notifications about changes.
* There was some chatter in this issue you can see in the edit history. I've [moved it](https://github.com/cisagov/manage.get.gov/issues/2370#issuecomment-2397710194) down below.

### Issue Links

- Related content ticket: #2197 
- blocked by #2597 
- relates to #3285 
- relates to #2960 
- bundled in #3296

**Comment by vickyszuchin** [comment]
Refinement label added because of comment in "Additional Context" section.

**Comment by vickyszuchin** [comment]
Discussed and refined in Discovery Fireside meeting (9/5).

**Comment by vickyszuchin** [comment]
Additional refinement questions added.

**Comment by h-m-f-t** [comment]
@Katherine-Osos and @witha-c to answer the questions and help us decide next steps.

**Comment by witha-c** [comment]
> @Katherine-Osos and @witha-c to answer the questions and help us decide next steps.

Done! Ready to be split into new tickets OR reviewed by product for agreement with our recommendations and then split.

**Comment by PaulKuykendall** [comment]
@abroddrick will make the precursor ticket to "Make the checkbox to disable email sending."

**Comment by h-m-f-t** [comment]
This content was primary comment in this issue, but I'm re-homing it here

> **Refinement Question:**
> 
> - We don't currently send an email if someone creates a domain invitation on stable for a user. If we send a domain invitation email for each domain invitation made in django admin, do you want us to update the the success message or add any info to let a user know that they will be sending a email by performing this action? Same goes for the deleting a User domain role (aka removing a user from a domain on django admin)
>   - **Design recommendation:** We should add a checkbox to the "add domain invitation" page in the admin that allows users to decide whether an email gets sent. This checkbox should be checked by default, meaning emails will be sent to the invitee and domain managers. However, if the box is unchecked by the user, then no emails will be sent. This should cover the messaging and inform the user of the action about to take place.
>  
> - There have been times when Cameron has had to make multiple domain invitations for a user, this will now send an email for each domain invitation manually created in admin. Do we want to instead give the analyst/superuser an option to send the email or not? Or are we okay that an analyst/superuser who makes multiple domains invitations in a short time span will end up sending multiple emails to a user. 
>   - **Design recommendation:** There is security in transparency and communication. Sending an email for each action should be the default. Our recommendation above to add a checkbox will allow analysts and superusers to choose to not send an email if there is a compelling reason. 
> - For removing a domain manager, if there is an error with the merge users feature we would need to manually remove a user with a deleted loging.gov account via django admin.  This would result in an email for each domain they are removed from. Then when they are re-added they will get an email for each domain sent to their new emails address inviting them to those domains. Has design/product considered that with this AC?
>     - **Design recommendation:** This is an example of a compelling reason to avoid all the emails. The checkbox will allow analysts and superusers to skip the email notifications for administrative moves like this.  
> - For indicating the email of who sent the domain invitation, what if it someone on django admin sending the domain invitation? Should we use help@get.gov as the email?  (Assuming we don't want to show analyst emails if they make an invitation)
>     - **Design recommendation:** yes, send it from help@get.gov

**Comment by katypies** [comment]
I think we still need a ticket to allow analysts to _not_ send an email when a domain manager is invited. @lizpearl  is going to create that ticket.

Since this is assigned to @abroddrick , there may be additional refinement needed, but we're not entirely sure what's missing, so need to get that documented and resolved with Alysia.

**Comment by h-m-f-t** [comment]
@katypies analysts can avoid sending an email by adding a user domain role directly, though I'm unsure where org model requests intersect with that.

**Comment by abroddrick** [comment]
@katypies the todo item I have for this ticket was to create a ticket for adding checkboxes so that analysts can turn this off if they want. That was going to block this one in particular. See Paul's comment here: https://github.com/cisagov/manage.get.gov/issues/2370#issuecomment-2383984244.

**Comment by Katherine-Osos** [comment]
@abroddrick Instead of checkboxes, we discussed a different approach when working on [another ticket](https://github.com/cisagov/manage.get.gov/issues/3019). Let's chat -- we can update this ticket together if that's helpful.

**Comment by abroddrick** [comment]
Per our convo today, @Katherine-Osos  I updated this one and created a separate ticket for t[he manager removal email ](https://github.com/cisagov/manage.get.gov/issues/3285). Please double check my attempt at the info messages. Leaving refinement on this for people to review the changes if needed. 

Also, Katherine (or anyone else) If the user is not part of a portfolio yet, and they are invited to a domain in a portfolio should we send a portfolio invitation? This becomes _potentially_  problematic as we are creating portfolios for FEB without them all being used. Which means if we did this (add a user to the portfolio too), they could get an invite to that portfolio and then not be able to see the portfolio when they log in. I would recommend against us inviting them to the portfolio too and instead we adjust the info message to indicate that the user will _not_ be added to the portfolio.

**Comment by Katherine-Osos** [comment]
@abroddrick I updated the description and added an AC for Django admin. In addition to this new email for existing domain managers, Django should also send the existing "domain invitation" email to the person who is being added as a manager. Take a look at the updated ACs and see if you're ok with that change.

**Comment by Katherine-Osos** [comment]
@abroddrick As for your second note about sending the portfolio invitation email ... if the domain is part of a portfolio, then the person needs to be added to the portfolio or the member data will be inaccurate for that organization. Maybe this is worth a discussion in Org Model meeting before working on this ticket. fyi, a similar AC is included in this [ticket](https://github.com/cisagov/manage.get.gov/issues/2960) for updating the domain invitation email and this [ticket](https://github.com/cisagov/manage.get.gov/issues/3019) for adding the portfolio invitation.

**Comment by abroddrick** [comment]
Keeping refinement on this to discuss those tickets in the org model meeting. @Katherine-Osos I will update your changes to reflect discussions had regarding keeping the sending domain invitation email as a seperate AC away from the status email
